internal/services/object/helpers_object.go

```diff
@@ -47,8 +47,23 @@ func (r *scalewayResolver) ResolveEndpoint(ctx context.Context, params s3.Endpoi
 	return s3.NewDefaultEndpointResolverV2().ResolveEndpoint(ctx, params)
 }
 
+type userAgentRoundTripper struct {
+	defaultRoundTripper http.RoundTripper
+	userAgent           string
+}
+
+func (r *userAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", r.userAgent)
+
+	return r.defaultRoundTripper.RoundTrip(req)
+}
+
 func newS3Client(ctx context.Context, region, accessKey, secretKey string, httpClient *http.Client) (*s3.Client, error) {
 	endpoint := "https://s3." + region + ".scw.cloud"
+	httpClient.Transport = &userAgentRoundTripper{
+		defaultRoundTripper: httpClient.Transport,
+		userAgent:           "Scaleway Terraform Provider",
```

Comment:
Could you add way more information `m/E aws-sdk-go-v2/1.32.3 os/macos lang/go#1.23.2 md/GOOS#darwin md/GOARCH#arm64 api/s3#1.66.2` The version of terraform ?

Reply:
Here is an example of the User agent we got:
```
            User-Agent:
                - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.22.2; darwin; amd64) terraform-provider/develop terraform/terraform-tests
```